packages/flutter/lib/src/foundation/stack_frame.dart

```diff
@@ -0,0 +1,230 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui' show hashValues;
+
+import 'package:meta/meta.dart';
+
+/// A object representation of a frame from a stack trace.
+///
+/// {@tool sample}
+///
+/// This example creates a traversable list of parsed [StackFrame] objects from
+/// the current [StackTrace].
+///
+/// ```dart
+/// final List<StackFrame> currentFrames = StackFrame.fromStackTrace(StackTrace.current);
+/// ```
+/// {@end-tool}
+@immutable
+class StackFrame {
+  /// Creates a new StackFrame instance.
+  ///
+  /// All parameters must not be null. The [className] may be the empty string
+  /// if there is no class (e.g. for a top level library method).
+  const StackFrame({
+    @required this.number,
+    @required this.column,
+    @required this.line,
+    @required this.packageScheme,
+    @required this.package,
+    @required this.packagePath,
+    this.className = '',
+    @required this.method,
+    this.isConstructor = false,
+    @required this.source,
+  })  : assert(number != null),
+        assert(column != null),
+        assert(line != null),
+        assert(method != null),
+        assert(packageScheme != null),
+        assert(package != null),
+        assert(packagePath != null),
+        assert(className != null),
+        assert(isConstructor != null),
+        assert(source != null);
+
+  /// A stack frame representing an asynchronous suspension.
+  static const StackFrame asynchronousSuspension = StackFrame(
+    number: -1,
+    column: -1,
+    line: -1,
+    method: 'asynchronous suspension',
+    packageScheme: '',
+    package: '',
+    packagePath: '',
+    source: '<asynchronous suspension>',
+  );
+
+  /// Parses a list of [StackFrame]s from a [StackTrace] object.
+  ///
+  /// This is normally useful with [StackTrace.current].
+  static List<StackFrame> fromStackTrace(StackTrace stack) {
+    assert(stack != null);
+    return fromStackString(stack.toString());
+  }
+
+  /// Parses a list of [StackFrame]s from the [StackTrace.toString] method.
+  static List<StackFrame> fromStackString(String stack) {
+    assert(stack != null);
+    return stack
+        .trim()
+        .split('\n')
+        .map(fromStackTraceLine)
+        .toList();
+  }
+
+  static StackFrame _parseWebFrame(String line) {
+    final bool hasPackage = line.startsWith('package');
+    final RegExp parser = hasPackage
+        ? RegExp(r'^(package:.+) (\d+):(\d+)\s+(.+)$')
+        : RegExp(r'^(.+) (\d+):(\d+)\s+(.+)$');
+    final Match match = parser.firstMatch(line);
+    assert(match != null, 'Expecgted $line to match $parser.');
```

Comment:
typo: Expected

Reply:
Saw this right after landing :( will open a PR to fix, thanks for catching that